examples/platforms/simulation/radio.c

```diff
@@ -299,25 +308,49 @@ static void ReverseExtAddress(otExtAddress *aReversed, const otExtAddress *aOrig
     }
 }
 
+static inline uint8_t getIidFromFrame(const otRadioFrame *aFrame)
+{
+    uint8_t iid = 0;
+    OT_UNUSED_VARIABLE(aFrame);
+#if OPENTHREAD_RADIO && OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE == 1
+    otPanId destPanId;
+
+    destPanId = otMacGetDstPanId(aFrame);
+    if (destPanId == RADIO_BCAST_PANID)
+    {
+        iid = RADIO_BCAST_IID;
+    }
+    else
+    {
+        iid = utilsSoftSrcMatchFindIidFromPanId(destPanId);
+    }
+#endif
+
+    return iid;
+}
+
 static bool hasFramePending(const otRadioFrame *aFrame)
 {
     bool         rval = false;
     otMacAddress src;
+    uint8_t      iid;
 
     otEXPECT_ACTION(sSrcMatchEnabled, rval = true);
     otEXPECT(otMacFrameGetSrcAddr(aFrame, &src) == OT_ERROR_NONE);
+    iid = getIidFromFrame(aFrame);
+    assert(iid != INVALID_VALUE);
```

Comment:
No need to assert here. There might be tests (now or in future), where we validate that packets with wrong dest panid are handled. We can either drop it or pass it with bcast iid.

Reply:
Agree, but since there is no test at this point for multipan for wrong dest panId I purposely added it to catch that unexpected frame. I thought we should remove this check whenever we purposely allow wrong dest panId, do we happen to know any use case that it might happen? I am happy to change it if it makes sense to remove this check though; let me know. I think simulated radio filtered the packets based on current and broadcast PANID so we should not be getting wrong dest panId unless radio in promiscuous mode.

For promiscuous mode, I think simulated radio allows all the packets but that code flow does not go through this function. 

For non multipan cases, iid will be always zero, so this should not affect the existing tests.

Reply:
Good point!!